keras/layers/core.py

```diff
@@ -16,21 +16,65 @@
 srng = RandomStreams(seed=np.random.randint(10e6))
 
 class Layer(object):
-    def __init__(self):
+    def __init__(self, name, prev, input_dim):
         self.params = []
+        self.name = name
+        self.prev = prev
+
+        if type(self.prev) is str:
+            self.prev = [self.prev] # single string or a list of strings under one op
+
+        self.input_dim = input_dim
+        self.output_dim = (None, )
+
+    def setup(self):
+        '''
+            Contains code that depends on values that are optional but can be understood from context (ex: input_dim)
+        '''
+        return
+
+    def connect(self, node):
+        self.previous = node
+
+        if self.input_dim == (None, ):
```

Comment:
What's the difference between `self.input_dim` being `(None,)` and `None`? Could it also be some combination of `None` and other values, such as `(128, None)`?